### DISCUSSION TITLE
Is there a kind of "sphlashscreen" to freeze entire UI when I load stuff?

### DISCUSSION THREAD

**Discussion by PawelRoman** [zauberzeug/nicegui#1988] [Q&A] [answered]
### Question

I'm building an app which makes calls to a REST API, and some of those calls may potentially take a few seconds to load some data. So, when a user clicks a button to get the data, I want to show them they need to wait. I don't want them to make subsequent clicks (or any other UI interactions), I just want the app to freeze the entire UI and wait until the data is loaded. 

Is there a component to do that? It would be ideal if it was possible to programmatically enable/disable something like a layer on top of all UI, with some kind of animated icon, like spinning circle and when it's enabled it would block the UI from any interactions.

Maybe it already exists? But I could not find anything like this in the docs.

thanks

**Comment by nibinho** [comment, marked as answer]
I think you could achieve this using ui.dialog and ui.card:
```python
import asyncio

from nicegui import ui


async def show_splashscreen():
    with ui.dialog(value=True).props(
        "persistent maximized"
    ) as dialog, ui.card().classes("bg-transparent"):
        ui.spinner(size="large")
        await asyncio.sleep(5) # <- logic goes in here
        dialog.close()


with ui.column():
    ui.button("Show Splashscreen", on_click=show_splashscreen)

ui.run()
```
